### ISSUE TITLE
[FEATURE REQ] Ability to use Default Subscription ID when creating management client

### ISSUE THREAD

**Issue by brandonh-msft** [Azure/azure-sdk-for-net#14548]
**Library or service name.**
`Azure.ResourceManager.Storage`
 
**Is your feature request related to a problem? Please describe.**
When using the Fluent API, I can do this:
```csharp
            return Microsoft.Azure.Management.Fluent.Azure
                .Authenticate(credentials)
                .WithDefaultSubscription();
```

Which picks the default subscription (presumably based on the given credentials); my application doesn't have to know its subscription id.

However, with the new libraries, the API (seemingly) requires subscription ID to create a new management client:

![image](https://user-images.githubusercontent.com/20270743/91219120-fd90d400-e6ce-11ea-9ffa-04bd97affefa.png)

which means I now have to have the Subscription ID for my application stored as an env var, etc in order to use the new management libraries. This will block us from using the new Management SDKs.

**Comment by jsquire** [comment]
Hi Brandon.   Thanks for opening this.  Tagging and routing to the team member best able to assist.

**Comment by blueww** [comment]
@brandonh-msft 
It seems your issue is for SDK [Microsoft.Azure.Management.Storage.Fluent](https://www.nuget.org/packages/Microsoft.Azure.Management.Storage.Fluent/).
If so, the project github repo for it is: https://github.com/Azure/azure-libraries-for-net , would you please raise your issue there?

**Comment by blueww** [comment]
@erich-wang , @pakrym
Would you please help to look at the “Azure.ResourceManager.Storage”  issue?
I see you have contribution to this lib.
Thanks!

**Comment by erich-wang** [comment]
This issue is assigned to @allenjzhang and he is right contact for addressing this request.

**Comment by blueww** [comment]
@erich-wang 
Thanks Erich! 
I will notice the issue since it have "Storage" tag, and system will notify me automatically. 
Not sure if we can add some tag to differ the Track1 SDK issue from Track2 SDK issue, then I can skip the track2 SDK issues.

**Comment by jsquire** [comment]
> Not sure if we can add some tag to differ the Track1 SDK issue from Track2 SDK issue, then I can skip the track2 SDK issues.

@blueww: This has been a subject of discussion amongst the triage team; currently, we've been asked not to make that distinction through tags.  The routing is intended to be done by ownership and not by generation of the SDK.  As second-level triage, Allen would make the determination for further routing and assign it to the correct individual or tag as `Service Attention` where appropriate.

**Comment by m-nash** [comment]
AB#4143

**Comment by riezebosch** [comment]
Looks like fixed (in 1.3.0) in `Azure.ResourceManager`:
```csharp
await new ArmClient(new DefaultAzureCredential()).GetDefaultSubscriptionAsync()
```

**Comment by ArthurMa1978** [comment]
This has been fixed in the stable release
